WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryViewModel.swift

```diff
@@ -0,0 +1,85 @@
+import Combine
+import Yosemite
+
+/// An item whose inventory can be displayed and managed
+///
+protocol InventoryItem {
+    var manageStock: Bool { get }
+    var stockQuantity: Decimal? { get }
+    var sku: String? { get }
+    var imageURL: URL? { get }
+
+    func retrieveName(with stores: StoresManager, siteID: Int64) async throws -> String
+}
+
+extension SKUSearchResult {
+    var inventoryItem: InventoryItem {
+        switch self {
+        case .product(let product):
+            return product
+        case .variation(let variation):
+            return variation
+        }
+    }
+}
+
+extension Product: InventoryItem {
+    func retrieveName(with stores: StoresManager, siteID: Int64) async throws -> String {
+        name
+    }
+}
+extension ProductVariation: InventoryItem {
+    func retrieveName(with stores: StoresManager, siteID: Int64) async throws -> String {
+        // Let's retrieve the parent product's name
+        return try await withCheckedThrowingContinuation { continuation in
+            let action = ProductAction.retrieveProduct(siteID: siteID,
+                                                       productID: productID) { result in
+                switch result {
+                case let .success(product):
+                    continuation.resume(with: .success(product.name))
+                case let .failure(error):
+                    continuation.resume(throwing: error)
+                }
+            }
+
+            Task { @MainActor in
+                stores.dispatch(action)
+            }
+        }
```

Comment:
Using checked continuations to call Actions in the WooCommerce target can make testing impossible, in my experience. 

I found a decent workaround, similar to what they've done on Android; create a super simple Service in Yosemite which wraps the actions in an async function. That wrapper still won't be testable itself, but the WooCommerce functions which call it can be. 

See `SystemStatusService` for an example...

Reply:
Thanks for the suggestion, I remember that case. I will follow the tests run on CI and adopt your strategy if they fail 👍